pykg2tbl/service.py

```diff
@@ -1,33 +1,41 @@
-# Use this file to describe the datamodel handled by this module
-# we recommend using abstract classes to achieve proper service and interface
-# insulation
 import csv
 import logging
 from abc import ABC, abstractmethod
+from typing import List, Union
 
+import pandas as pd
 from rdflib import Graph
 from SPARQLWrapper import SPARQLWrapper
 
 log = logging.getLogger(__name__)
 
 
-class QueryResult:
+class QueryResult(ABC):
     """
     Class that incompases the result from a performed query
 
-    :param data: query result data in the form of a list of dictionaries
+    :param list data: query result data in the form of a list of dictionaries
+    :param str query: query
+
     """
 
-    def __init__(self, data: dict):
-        log.info(data)
-        self._data = data  # TODO pandas dataframe
+    def __init__(self, data: list, query: str = ""):
+        # log.info(data)
+        self._data = data
+        self.query = query
 
     def __str__(self):
         # TODO consider something smarter then this:
         return str(self._data)
```

Comment:
@marc-portier @cedricdcc  what is the idea where?
What should be the output of str() ?

Reply:
As it is now it will be a dump of the raw dict but as string. I think this doesn't have a lot of worth atm since I don't think it is used anywhere in the code.

Reply:
I will rephrase the question.
Is this worth keeping?
If so, what is the idea here? I would guess not just the raw output, right?

Reply:
I think you should ask @marc-portier that question. I have no intent in keeping the __str__

Reply:
having a `__str__` implementation is good practice imho

now that we (recently made this QueryResult an AbstractBaseClass there is not really anything useful you can do I suppose

for its subclasses I think it does make sense to make something that gives a meaningful output when used in print() or log statement